src/org/labkey/test/components/domain/RangeValidatorPanel.java

```diff
@@ -0,0 +1,207 @@
+                    "validator panel did not become expanded", 1000);
+        }
+        return this;
+    }
+
+    public boolean isExpanded()
+    {
+        return !elementCache().collapseIconLocator.existsIn(this);
+    }
+
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+
+    protected class ElementCache extends WebDriverComponent.ElementCache
+    {
+
+        public Select firstConditionSelect()        // these elements are transient; when the panel is collapsed they disappear
+        {
+            return SelectWrapper.Select(
+                    Locator.tagWithAttribute("select", "name", "domainpropertiesrow-firstFilterType"))
+                    .waitFor(this);
+        }
+        public Input firstFilterValueInput()
+        {
+            return Input(Locator.input("domainpropertiesrow-firstFilterValue"), getDriver()).waitFor(this);
+        }
+
+        public Select secondConditionSelect()
+        {
+            return SelectWrapper.Select(
+                    Locator.tagWithAttribute("select", "name", "domainpropertiesrow-secondFilterType"))
+                    .waitFor(this);
+        }
+        public Input secondFilterValueInput()
+        {
+            return Input(Locator.input("domainpropertiesrow-secondFilterValue"), getDriver()).waitFor(this);
+        }
+
+        public Input descriptionInput()
+        {
+            return Input(Locator.textarea("domainpropertiesrow-description"), getDriver()).waitFor(this);
+        }
+        public Input errorMessageInput()
+        {
+            return Input(Locator.textarea("domainpropertiesrow-errorMessage"), getDriver()).waitFor(this);
+        }
+        public Input nameInput()
+        {
+            return Input(Locator.input("domainpropertiesrow-name"), getDriver()).waitFor(this);
+        }
+
+        final Locator collapseIconLocator = Locator.tagWithClass("div", "domain-validator-collapse-icon");
+        final WebElement removeButton = Locator.button("Remove Validator").findWhenNeeded(this);
+    }
+
+
+    public static class RangeValidatorPanelFinder extends WebDriverComponentFinder<RangeValidatorPanel, RangeValidatorPanelFinder>
+    {
+        private Locator.XPathLocator _baseLocator = Locator.tagWithClass("div", "domain-validator-panel");
+        private String _name = null;
+        private String _id = null;
+
+        public RangeValidatorPanelFinder(WebDriver driver)
+        {
+            super(driver);
+        }
+
+        public RangeValidatorPanelFinder closedByName(String name)          //finds
+        {
+            _name = name;
+            return this;
+        }
+        public RangeValidatorPanelFinder openedByName(String name)          //finds the currently-expanded
+        {
+            _baseLocator = _baseLocator.withDescendant(Locator.input("domainpropertiesrow-name")
+                    .withAttribute("value", name));
+            return this;
+        }
```

Comment:
Why combine the name and open-ness instead of providing a separate method for each?
e.g.:
`public RangeValidatorPanelFinder withName(String name)`
-and-
`public RangeValidatorPanelFinder opened()`

Reply:
...it's because the way to find it by name is different when opened or closed.
When open, the name is in the name field edit.  When closed, it is part of the expression summary (which disappears when opened)